service/qos/service_level_controller.cc

```diff
@@ -298,6 +298,22 @@ void service_level_controller::update_from_distributed_data(std::chrono::duratio
     }
 }
 
+future<> service_level_controller::stop_update_from_distributed_data() {
+    if (this_shard_id() != global_controller) {
+        return make_ready_future();
+    }
+    // abort the loop of the distributed data checking if it is running
+    if (!_global_controller_db->distributed_data_update.available()) {
```

Comment:
I think this check is also not really needed - requesting an abort if this future is ready should have no effect on anything anyway, so we may execute `request_abort` unconditionally

Reply:
Except it will invalidate the abort source (which I will have to recreate), unless we decide that stop always get **called** only 
after start and you can never restart this loop until you restart Scylla.